nvtabular/columns/schema.py

```diff
@@ -36,7 +137,32 @@ def with_tags(self, tags) -> "ColumnSchema":
 
         tags = list(set(list(self.tags) + tags))
 
-        return ColumnSchema(self.name, tags=tags)
+        return ColumnSchema(self.name, tags=tags, properties=self.properties)
+
+    def with_properties(self, properties):
+        if not isinstance(properties, dict):
+            raise TypeError("properties must be in dict format, key: value")
+
+        # Using new dictionary to avoid passing old ref to new schema
+        properties.update(self.properties)
+
+        return ColumnSchema(self.name, tags=self.tags, properties=properties)
+
+    def __eq__(self, other):
+        if not isinstance(other, ColumnSchema):
+            return False
+        if (
+            self.name == other.name
+            and self.tags == other.tags
+            and len(other.properties) == len(self.properties)
+            and self._is_list == other._is_list
```

Comment:
Should we be using the dtype here too ?  If so, can we just use the default ```__eq__``` method provided by the dataclass ?